docs/understanding-airbyte/jobs.md

```diff
@@ -1,45 +1,111 @@
 # Workers & Jobs
 
-In Airbyte, all interactions with connectors are run as jobs performed by a Worker. Examples of workers are:
+In Airbyte, all interactions with connectors are run as jobs performed by a Worker. Each job has a corresponding worker:
 
 * Spec worker: retrieves the specification of a connector \(the inputs needed to run this connector\)
 * Check connection worker: verifies that the inputs to a connector are valid and can be used to run a sync
 * Discovery worker: retrieves the schema of the source underlying a connector
 * Sync worker, used to sync data between a source and destination
 
+Thus, there are generally 4 types of workers.
+
+**Note: Workers here refers to Airbyte workers. Temporal, which Airbyte uses under the hood for scheduling, has its own worker concept. This distinction is important.**
+
+## Job State Machine
+
+Jobs have the following state machine.
+
+![Job state machine](../.gitbook/assets/job-state-machine.png)
+
+[Image Source](https://docs.google.com/drawings/d/1cp8LRZs6UnhAt3jbQ4h40nstcNB0OBOnNRdMFwOJL8I/edit)
+
 ## Worker Responsibilities
 
-The worker has 4 main responsibilities in its lifecycle. 
+The worker has the following responsibilities.
+
+1. Handle the process lifecycle for job-related processes. This includes starting, monitoring and shutting down processes.
+2. Facilitate message passing to or from various processes, if required. \(more on this [below](jobs.md#worker-types)\).
+3. Handle job-relation operational work such as:
+   1. Basic schema validation.
+   2. Returning job output, including any error messages. \(See [Airbyte Specification](airbyte-protocol.md) to understand the output of each worker type.\)
+   3. Telemetry work e.g. tracking the number and size of records within a sync.
 
-1. Spin up any connector docker containers that are needed for the job. 
-2. They facilitate message passing to or from a connector docker container \(more on this [below](jobs.md#message-passing)\). 
-3. Shut down any connector docker containers that it started. 
-4. Return the output of the job. \(See [Airbyte Specification](airbyte-protocol.md) to understand the output of each worker type.\)
+Conceptually, **workers contain the complexity of all non-connector-related job operations**. This lets each connector be as simple as possible.
 
-## Message Passing
+### Worker Types
 
 There are 2 flavors of workers: 
 
-1. There are workers that interact with a single connector \(e.g. spec, check, discover\) 
-2. There are workers that interact with 2 connectors \(e.g. sync, reset\)
+1. **Synchronous Job Worker** - Workers that interact with a single connector \(e.g. spec, check, discover\).
```

Comment:
Note for later, I find our Sync/Async naming for the workers a bit confusing. Especially when you run into the `SyncWorkflow` which I think is doing the work for the Async Worker.

Reply:
haha good point! Do you think it's worth using another term?

Reply:
I would stick to what we have in the code for now. That's an item I'd want to rename eventually, right now, I am mostly trying to stay clear of the partial rename which introduces even more confusion.

Reply:
Ok. Let's keep this for now. I think it's more valuable to get this documented. We can follow up on naming later.